node/node.go

```diff
@@ -503,7 +531,12 @@ func (n *Node) Restart() error {
 func (n *Node) Attach() (*rpc.Client, error) {
 	n.lock.RLock()
 	defer n.lock.RUnlock()
+	return n.AttachLocked()
+}
 
+// AttachLocked creates an RPC client attached to an in-process API handler.
+// Note, this function assumes the lock of node is held.
+func (n *Node) AttachLocked() (*rpc.Client, error) {
```

Comment:
Since we need to attach the local node during the node `Start` phase, but the lock is already held at the beginning of the `Start` function. So I think we should expose this function(I agree it's weird and unsafe) anyway unless we move the callback invocation out of `Start` function.